homeassistant/components/binary_sensor/ihc.py

```diff
@@ -3,60 +3,39 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/binary_sensor.ihc/
 """
-import voluptuous as vol
-
 from homeassistant.components.binary_sensor import (
-    BinarySensorDevice, PLATFORM_SCHEMA, DEVICE_CLASSES_SCHEMA)
+    BinarySensorDevice)
 from homeassistant.components.ihc import (
-    validate_name, IHC_DATA, IHC_CONTROLLER, IHC_INFO)
-from homeassistant.components.ihc.const import CONF_INVERTING
+    IHC_DATA, IHC_CONTROLLER, IHC_INFO)
+from homeassistant.components.ihc.const import (
+    CONF_INVERTING)
 from homeassistant.components.ihc.ihcdevice import IHCDevice
 from homeassistant.const import (
-    CONF_NAME, CONF_TYPE, CONF_ID, CONF_BINARY_SENSORS)
-import homeassistant.helpers.config_validation as cv
+    CONF_TYPE)
 
 DEPENDENCIES = ['ihc']
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_BINARY_SENSORS, default=[]):
-        vol.All(cv.ensure_list, [
-            vol.All({
-                vol.Required(CONF_ID): cv.positive_int,
-                vol.Optional(CONF_NAME): cv.string,
-                vol.Optional(CONF_TYPE): DEVICE_CLASSES_SCHEMA,
-                vol.Optional(CONF_INVERTING, default=False): cv.boolean,
-            }, validate_name)
-        ])
-})
-
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the IHC binary sensor platform."""
-    ihc_controller = hass.data[IHC_DATA][IHC_CONTROLLER]
-    info = hass.data[IHC_DATA][IHC_INFO]
     devices = []
-    if discovery_info:
+    if discovery_info:  
```

Comment:
Move this to the beginning of the function body, invert the check and return if true:
```py
if discover_info is None:
    return
```
We will save indentation that way.